### ISSUE TITLE
Connection load fails

### ISSUE THREAD

**Issue by davidsoderberg** [doug-martin/nestjs-query#296]
When I try this query
```{
  posts {
    edges {
      node {
        id
        categories {
          edges {
            node {
              id
            }
          }
        }
      }
    }
  }
}
```
And I get:
```
"stacktrace": [
            "TypeError: Right-hand side of 'instanceof' is not an object",
            "    at DefaultAssembler.toPlain (/Users/user/Code/project-api/node_modules/@nestjs-query/core/src/assemblers/class-transformer.assembler.ts:34:20)",
            "    at DefaultAssembler.convertToDTO (/Users/user/Code/project-api/node_modules/@nestjs-query/core/src/assemblers/class-transformer.assembler.ts:12:45)",
            "    at entities.map (/Users/user/Code/project-api/node_modules/@nestjs-query/core/src/assemblers/abstract.assembler.ts:43:37)",
            "    at Array.map (<anonymous>)",
            "    at DefaultAssembler.convertToDTOs (/Users/user/Code/project-api/node_modules/@nestjs-query/core/src/assemblers/abstract.assembler.ts:43:21)",
            "    at entityRelations.reduce (/Users/user/Code/project-api/node_modules/@nestjs-query/query-typeorm/src/services/relation-query.service.ts:185:32)",
            "    at Array.reduce (<anonymous>)",
            "    at TypeOrmQueryService.batchQueryRelations (/Users/user/Code/project-api/node_modules/@nestjs-query/query-typeorm/src/services/relation-query.service.ts:183:28)",
            "    at process._tickCallback (internal/process/next_tick.js:68:7)"
          ]
```
this is how my module config looks like:
```
{
  imports: [NestjsQueryTypeOrmModule.forFeature(entities)],
  assemblers: [PostAssembler, CategoryAssembler],
  resolvers: [
    {
      DTOClass: Post,
      EntityClass: Post,
      AssemblerClass: PostAssembler
    },
    {
      DTOClass: Category,
      EntityClass: Category,
      AssemblerClass: CategoryAssembler
    }
  ]
};
```
What have I done wrong?

**Comment by davidsoderberg** [comment]
I figured out what it is, this framework supports
`@ManyToMany(() => Category, 'posts')`
but not:
`@ManyToMany('Category', 'posts')`
For typeorm, both of the are supported by typeorm so maybe this framework should to?

**Comment by doug-martin** [comment]
@davidsoderberg can you provide an example of your DTO/Entity so I can see how you set up the connection.

**Comment by doug-martin** [comment]
@davidsoderberg this is because the relation resolver does not check if the relation is defined and I am hesitant to do since the value could be an empty array, promise etc depending on the ORM and how the entity is defined. 

However, you can override the `queryBooks` method to add your custom logic.

**NOTE** The `queryBooks` method will be called for both query (`authors`) and findById (`author`) endpoints 

```ts
import { QueryService, InjectQueryService } from '@nestjs-query/core'
import { ConnectionType, CRUDResolver, CursorPagingType, PagingStrategies } from '@nestjs-query/query-graphql';
import { Resolver, Args, Int, Query, ResolveField, Parent } from '@nestjs/graphql';
import { Author } from './author.dto'
import { AuthorEntity } from './author.entity'
import { Book } from '../book/book.dto';
import { ExecutionContext } from '@nestjs/common';
import { QueryType } from '@nestjs-query/query-graphql/dist/src/types/query/query-args';

@Resolver(() => Author)
export class AuthorResolver extends CRUDResolver(Author, {
  read: { one: { disabled: true } }, // disabling read one
}) {
  constructor(
    @InjectQueryService(AuthorEntity)
    readonly service: QueryService<AuthorEntity>,
  ) {
    super(service)
  }


  // returning plain JSON
  // I want to override the default (read one) resolver, and I dont want to invoke any database request.
  @Query(() => Author)
  async author(@Args('id', { type: () => Int }) id: number) {
    console.log('custom resolver running')
    return {
      "id": 7,
      "email": "Hortense84@hotmail.com",
      "firstName": "Adonis",
      "lastName": "Boyle",
      // the "books" part is being resolved somewhere which create a database request.
      // It would be awesome if there is a way to disable it and just return the nested data already defined in here.
      "books": {
        "edges": [
          {
            "node": {
              "id": 6,
              "isPublished": false,
              "title": "a exercitationem atque",
              "summary": "Corporis magni in quod consequatur dolor aut atque veritatis. Dolor similique quisquam facilis quae minima ducimus. Mollitia eos nam ratione aliquid tempora dolore. Eveniet possimus ex. Omnis nam placeat.",
              "publishedDate": "2020-02-13"
            }
          },
        ]
      }
    }
  }

  @ResolveField('books')
  async queryBooks(@Parent() author: Author & {books?: ConnectionType<Book>}, query: QueryType<Book, CursorPagingType>, ctx: ExecutionContext): Promise<ConnectionType<Book>> {
    if(author.books){
      return author.books;
    }
    // @ts-ignore
    return super.queryBooks(author, query, ctx);
  }
}

```

**Comment by doug-martin** [comment]
Sorry commented on wrong issue :(

**Comment by davidsoderberg** [comment]
Ok, this is my simplest example:
Post.ts:
```
@ManyToMany('Category', 'posts')
categories: Category[];
```
Category.ts:
```
@ManyToMany('Post', 'categories')
@JoinTable()
posts: Post[];
```

And the connection looks like this:
`@Connection('categories', () => Category)`

Please tell me if you need more?

**Comment by davidsoderberg** [comment]
I can try to fix it in a pr before monday, if you like?

**Comment by doug-martin** [comment]
@davidsoderberg I have figure out the problem, just working on tests

**Comment by doug-martin** [comment]
@davidsoderberg just published at fix to `v0.14.2`!
